mem.go

```diff
@@ -265,6 +265,17 @@ func NewReader(m RO) *Reader {
 	return &Reader{sr: strings.NewReader(m.str())}
 }
 
+// Cut works like strings.Cut, but takes and returns ROs.
+func Cut(m, sep RO) (before, after RO, ok bool) {
```

Comment:
I’ve also had cause to implement a `CutLast` in parallel to `Index` vs `IndexLast`. I know it’s not in the builtin `strings` package, but something to consider.